react-native-community/cli
# feat: introduce /events endpoint for devtools like Flipper

Summary:
---------

This PR introduce a new endpoint, `/events` to be used by developer tooling, that services two goals:
1. Stream the applications JavaScript logs and metro events
2. Support receiving commands and forward them to Metro. The setup is generic, but the two use cases for now are "reload" and "open dev menu"

This change reflects changes we made to the CLI at FB as well. This enables Flipper to have better first class support for React Native. Although some Flipper plugins could already be used on RN through the generic ADB support, this change makes it possible to see the Metro server as a real device and interact with it. 

Preview:

<img width="1458" alt="Screen Shot 2020-02-10 at 17 51 05" src="https://user-images.githubusercontent.com/1820292/74227021-79887900-4cb5-11ea-86cf-ff9018f1f7a3.png">

https://youtu.be/HIau7zjJ3gg

Note that currently errors are not send by RN to the Metro logging, so in the current RN version those won't be visible, but that will change in a next minor or patch release in RN.

Test Plan:
----------

1. Build the repo
2. Take a test project and update the CLI according to the contributing guidelines. 
3. Start Flipper (the supporter Flipper version didn't have an OSS release yet so need to clone from source and `yarn && yarn start`)
4. Start the RN app with `yarn android` (or `ios`)
4. React Native should show up as device, logging, open devtools and reload should work
